.github/scripts/support/monday.js

```diff
@@ -956,12 +921,44 @@ module.exports = function Monday(issue) {
     );
   }
 
+  /**
+   * Set the issue status to "Unassigned" or "Assigned" based on assigned status.
+   * Default condition is: if issue is open, not in lifecycle, and not in a milestone status.
+   * @param {object} [params]
+   * @param {boolean} [params.assignedCondition] - Condition to set status to "Assigned"
+   * @param {boolean} [params.unassignedCondition] - Condition to set status to "Unassigned"
+   * @returns {void}
+   */
+  function setAssignedStatus({ assignedCondition, unassignedCondition } = {}) {
+    const ASSIGNED = "Assigned";
+    const UNASSIGNED = "Unassigned";
+    const defaultCondition =
+      issue.state === "open" &&
+      notInLifecycle({ labels }) &&
+      !inMilestoneStatus();
+    const shouldSetAssigned =
+      assignedCondition !== undefined ? assignedCondition : defaultCondition;
+    const shouldSetUnassigned =
+      unassignedCondition !== undefined
+        ? unassignedCondition
+        : defaultCondition;
+
+    if (shouldSetAssigned && assignee) {
+      setColumnValue(columnIds.status, ASSIGNED);
+      console.log("Status set to 'Assigned'.");
```

Comment:
You could DRY this up a bit by reusing the assignment variable values in string templates for these log messages.